test/ccall.jl

```diff
@@ -123,7 +126,11 @@ function test_struct1{Struct}(::Type{Struct})
     b = Float32(123.456)
 
     a2 = copy(a)
-    x = ccall((:test_1, libccalltest), Struct, (Struct, Float32), a2, b)
+    if Struct === Struct1
```

Comment:
FunctionWrappers.jl already doesn't like the current situation (https://github.com/yuyichao/FunctionWrappers.jl/blob/master/src/FunctionWrappers.jl#L77), so it won't be affected at all. The only change this makes is that now the argument type tuple ABI must be actually statically computable from the signature of the containing function (using `@generated` qualifies for this), instead of being required to be statically computable from "whatever we could constant propagate during codegen + some semi-arbitrary heuristics".

Don't worry, this is quite non-breaking in practice. Really it's just a simplification of the rule that the argument type tuple must be a constant literal. In the future, it might even allow us to relax that constraint (changing from literal -> evaluated at the toplevel during method definition).